pydantic/_internal/_generate_schema.py

```diff
@@ -248,8 +252,16 @@ def _generate_schema_from_property(self, obj: Any, source: Any) -> core_schema.C
         """
         get_schema = getattr(obj, '__get_pydantic_core_schema__', None)
         if get_schema is not None:
+            # (source) -> CoreSchema
+            if len(inspect.signature(get_schema).parameters) == 1:
+                return get_schema(source)
+
+            # (source, handler) -> CoreSchema
+            def handler(source: Any) -> core_schema.CoreSchema:
```

Comment:
Here handler accepts a source: Any, but down below, the annotations are:
```python
    @classmethod
    def __get_pydantic_core_schema__(
        cls, source: type[Any], handler: Callable[[], core_schema.CoreSchema]
    ) -> core_schema.CoreSchema:
```
Note — handler doesn't accept arguments. This seems wrong to me, it seems it `handler` should accept a type[Any] and return a `core_schema.CoreSchema`.